java/kotlin-extractor/deps.bzl

```diff
@@ -0,0 +1,126 @@
+load("//java/kotlin-extractor:versions.bzl", "DEFAULT_FALLBACK_VERSION", "VERSIONS", "version_less")
+load("//misc/bazel:lfs.bzl", "lfs_smudge")
+
+_kotlin_dep_build = """
+load("@rules_kotlin//kotlin:jvm.bzl", "kt_jvm_import")
+
+kt_jvm_import(
+    name = "{name}",
+    jar = "{name}.jar",
+    visibility = ["//visibility:public"],
+)
+"""
+
+_empty_zip = "PK\005\006\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000"
+
+def _get_dep(repository_ctx, name):
+    return repository_ctx.path(Label("//java/kotlin-extractor/deps:%s" % name))
+
+def _kotlin_dep_impl(repository_ctx):
+    _, _, name = repository_ctx.name.rpartition("~")
+    lfs_smudge(repository_ctx, [_get_dep(repository_ctx, name + ".jar")])
+
+    # for some reason rules_kotlin warns about these jars missing, this is to silence those warnings
+    repository_ctx.file("empty.zip", _empty_zip)
+    for jar in (
+        "annotations-13.0.jar",
+        "kotlin-stdlib.jar",
+        "kotlin-reflect.jar",
+        "kotlin-script-runtime.jar",
+        "trove4j.jar",
+    ):
+        repository_ctx.symlink("empty.zip", jar)
+    repository_ctx.file("BUILD.bazel", _kotlin_dep_build.format(name = name))
+
+_kotlin_dep = repository_rule(
+    implementation = _kotlin_dep_impl,
+)
+
+def _walk(dir):
+    res = []
+    next_dirs = [dir]
+
+    # loops must be bounded in starlark
```

Comment:
and we can't use some hacky `native.glob` to enumerate a (superset of) what we want to walk?

Reply:
I think `glob` is not available in repository rules

Reply:
yep, no glob available in repository rules (or even normal rules, only in macros).